Lib/test/support/__init__.py

```diff
@@ -377,6 +377,21 @@ def wrapper(*args, **kw):
     return decorator
 
 
+def thread_unsafe(reason):
+    """Mark a test as  not thread safe. When the test runner is run with
+    --parallel-threads=N, the test will be run in a single thread."""
+    def decorator(test_item):
+        test_item.__unittest_thread_unsafe__ = True
+        # the reason is not currently used
+        test_item.__unittest_thread_unsafe__why__ = reason
+        return test_item
+    if isinstance(reason, types.FunctionType):
+        test_item = reason
+        reason = ''
+        return decorator(test_item)
```

Comment:
Not *that* fond of making things "do what I mean" this way, but I guess passing reason is rare enough?

```suggestion
        return decorator(reason)
```

Reply:
That doesn't quite work because `reason` is also referenced in the `decorator`. The whole thing is kind of ugly, but it matches `unittest.skip` in behavior and in code structure:

https://github.com/python/cpython/blob/285c1c4e9543299c8bf69ceb39a424782b8c632e/Lib/unittest/case.py#L148-L166